### PR TITLE
Arceus Set Bugfixes

### PR DESCRIPTION
**Pull request by luongthomas**
Going through the bugs from trello regarding gen4 Arceus cards

### CODE REVIEW

**Comment by luongthomas** [comment]
[BAGON_SH10](https://trello.com/c/vxhJztmb/210-bagonsh10)

**Comment by luongthomas** [comment]
[SHINX_SH12](https://trello.com/c/w7tMgW4R/212-shinxsh12)

**Comment by luongthomas** [comment]
[DEPARTMENT_STORE_GIRL_85](https://trello.com/c/DSa4mQoI/195-departmentstoregirl85)

**Comment by luongthomas** [comment]
[PIKACHU_71](https://trello.com/c/aeUh6fSO/181-pikachu71)

**Comment by luongthomas** [comment]
[ELECTRIKE_63](https://trello.com/c/Y9ZZbIHi/173-electrike63)

**Comment by luongthomas** [comment]
[GASTLY_64](https://trello.com/c/y5cgr6zO/174-gastly64)

**Comment by luongthomas** [comment]
[GASTLY_36](https://trello.com/c/gYkiwfP6/146-gastly36)

**Comment by luongthomas** [comment]
[ELECTRIKE_62](https://trello.com/c/3Kz298gn/172-electrike62)

**Comment by luongthomas** [comment]
[FROSLASS_2](https://trello.com/c/bgZuQKoB/112-froslass2)

**Comment by luongthomas** [comment]
[AERODACTYL_13](https://trello.com/c/JNTyUJOP/122-aerodactyl13)

**Comment by luongthomas** [comment]
[PONYTA_46](https://trello.com/c/ANKWNMNl/156-ponyta46)

**Comment by luongthomas** [comment]
[CHARIZARD_1](https://trello.com/c/IQLOZQF8/111-charizard1)

**Comment by luongthomas** [comment]
[PICHU_25](https://trello.com/c/O6dB0aQZ/135-pichu25)

**Comment by luongthomas** [comment]
[HELIX_FOSSIL_93](https://trello.com/c/g0QtwtE7/203-helixfossil93)

**Comment by luongthomas** [comment]
[DOME_FOSSIL_92](https://trello.com/c/bu84Xew1/202-domefossil92)

**Comment by luongthomas** [comment]
[ARCEUS_AR4](https://trello.com/c/Rr1ygxSs/213-arceusar4)

**Comment by luongthomas** [comment]
[BEGINNING_DOOR_82](https://trello.com/c/fDkEc17L/192-beginningdoor82)

**Comment by luongthomas** [comment]
[REGISTEEL_39](https://trello.com/c/pQq7ZTa1/1144-registeel39)

File: src/tcgwars/logic/impl/gen4/Arceus.groovy [hunk 1]
```diff
@@ -248,6 +248,7 @@ public enum Arceus implements LogicCardInfo {
               if (r==PLAY_FROM_HAND && my.deck && confirm("Use Snow Gift")) {
                 powerUsed()
                 my.deck.search(min:1,"Search your deck for a card",{true}).moveTo(hidden:true,my.hand)
+                shuffleDeck()
```
Comment (flagrama) [inline src/tcgwars/logic/impl/gen4/Arceus.groovy:251]: Is there ever a situation where a `deck.search` should not shuffle? I wonder if this should be added to the `search` method.

**Review by flagrama — APPROVED** [review]

File: src/tcgwars/logic/impl/gen4/Arceus.groovy (hunk 1 above)
Comment (luongthomas) [reply src/tcgwars/logic/impl/gen4/Arceus.groovy:251]: That's a good point, I'll look for instances where there isn't a need to shuffle after a search

**Pull request merged**